### PR TITLE
42114 Enable Private Endpoint to Event Grid Domain

### PR DESCRIPTION
**Pull request by amogh14893**
PR contains below changes

1. Terraform Module for Private Endpoint 
2. Private DNS zone configuration is done by DDC post Private Endpoint deployment hence changes have been done for terraform state file to ignore private DNS zone configuration during deployment.
3.Changes for enabling Private Endpoint in event grid domain under networking.

### CODE REVIEW

File: Deployment/src/main.tf [hunk 1]
```diff
@@ -107,3 +113,13 @@ module "storage" {
   env_name            = local.env_name
   service_name        = local.service_name
 }
+
+module "PrivateEndPoint" {
+  source              = "./Modules/PrivateEndPoint"
```
[the remark this answers is not in the document]
Comment (amogh14893) [reply Deployment/src/main.tf]: Suggested changes applied

File: Deployment/templates/continuous-deployment.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ steps:
       TF_VAR_agent_vnet_name: $(agentVnetName)
       TF_VAR_agent_subnet_name: $(agentSubnetName)
       TF_VAR_agent_subscription_id: $(agentSubscriptionId)
+      TF_VAR_private_endpoint_subnet_name: $(privateEndpointSubnet)
+      TF_VAR_private_dns_zone_resource_id: $(PrivateDNSZoneId)
```
[the remark this answers is not in the document]
Comment (amogh14893) [reply Deployment/templates/continuous-deployment.yml]: We need **TF_VAR_private_endpoint_subnet_name** as this was created by DDC without delegation to refer in the Private end point.
TF_VAR_private_dns_zone_resource_id is not needed as Private DNS Zone is manually created

**Review by vijendr13545 — APPROVED** [review]
Looks good to me. Thanks.

**Comment by barrie-cooper** [comment]
See [Event Grid Domain Private Endpoint setup](https://dev.azure.com/ukhydro/External%20Notification%20Service/_wiki/wikis/External-Notification-Service.wiki/201/Event-Grid-Domain-Private-Endpoint-setup) for details of the manual setup that goes along with this.

**Review by barrie-cooper — APPROVED** [review]
LGTM

**Pull request merged**